### PR TITLE
[Enhancement] Support refresh iceberg mv by partition

### PR DESCRIPTION
**Pull request by Youngwb**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 87 / 110 (79.09%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergPartitionUtils.java|38|57|66.67%|[37, 52, 53, 55, 56, 58, 59, 60, 65, 96, 97, 98, 99, 100, 101, 104, 105, 114, 115]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|25|28|89.29%|[714, 730, 735]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|16|17|94.12%|[1681]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 102 / 124 (82.26%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|1|2|50.00%|[697]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergPartitionUtils.java|45|57|78.95%|[37, 56, 96, 97, 98, 99, 100, 101, 104, 105, 114, 115]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|23|29|79.31%|[1224, 1262, 1292, 1322, 1353, 1694]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|25|28|89.29%|[742, 758, 763]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        Set<IcebergPartition> changedPartition = getChangedPartition(table, from);
+        return changedPartition.stream().map(partition -> PartitionUtil.
+                convertIcebergPartitionToPartitionName(table.spec(), partition.data)).collect(Collectors.toSet());
+    }
+
+    public static Set<IcebergPartition> getChangedPartition(Table table, long from) {
+        ImmutableSet.Builder<IcebergPartition> builder = ImmutableSet.builder();
+        Snapshot snapShot = table.currentSnapshot();
+        if (snapShot.timestampMillis() >= from) {
+            while (snapShot.parentId() != null) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java:79]: what does this mean ?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                snapShot = table.snapshot(snapShot.parentId());
+                if (snapShot.timestampMillis() <= from) {
+                    break;
+                }
+            }
+            if (snapShot.timestampMillis() <= from) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java:85]: why <=

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java (hunk 1 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java:79]: better to put this logic into a function named getLastestSnapshot?

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java:79]: It means find the previous snapshot

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java (hunk 2 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java:85]: find the snapshot which timestamp is **from** or snapshot which timestamp is before **from**

**Review by murphyatwork — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by Youngwb** [comment]
@Mergify backport branch-3.2

**Comment by Youngwb** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**